### ISSUE TITLE
Link invoice positions of supplier invoices to projects

### ISSUE THREAD

**Issue by Natureshadow** [Dolibarr/dolibarr#23229]
### Feature Request

I would like to be able to link single invoice positions of supplier invoices to different projects, instead of the whole invoice.

### Use case

Some suppliers, like Hetzner, are unable (or unwilling) to send separate invoices per project. In order to have a better overview of expenses per project, and still have one supplier invoice as one accountancy booking, it is thuse necessary to split the invoice among projects by linking separate positions.

**Comment by Natureshadow** [comment]
Still relevant.

**Comment by camlafit** [comment]
Hello

I confirm could be interesting to associate partial invoice (customer or provider) to a project.
In some case we can get a global invoice and need to affect to different project some lines.

**Comment by hregis** [comment]
@Natureshadow @camlafit yes, but how can we quantify the distribution of the bill between the projects?

**Comment by Natureshadow** [comment]
>@Natureshadow @camlafit yes, but how can we quantify the distribution of the bill between the projects?

Not sure whether I get the question right, but... sum up the positions linked to the project 🤔?

Look at the original proposal; it proposes linking invoice positions, not some random split amounts.

**Comment by camlafit** [comment]
Hello

As actually we can affect a project to invoice/any object, could be set to line object sublevel.
If we can target line by line we can quantify the distribution of the bills.
Affect invoice to a project is as affect all lines. 

If some lines are not associated, should be not to care in consideration by project.

**Comment by JonBendtsen** [comment]
I think this can be done with Expense reports?

**Comment by JonBendtsen** [comment]
I still think it can be done with expense reports
